src/gl_fns.rs

```diff
@@ -129,6 +129,43 @@ impl Gl for GlFns {
         }
     }
 
+    fn read_pixels_into_buffer_with_stride(
+        &self,
+        x: GLint,
+        y: GLint,
+        width: GLsizei,
+        height: GLsizei,
+        byte_stride: GLsizei,
+        format: GLenum,
+        pixel_type: GLenum,
+        dst_buffer: &mut [u8],
+    ) {
+        let bpp = bpp(format, pixel_type);
+        assert!(byte_stride % bpp == 0);
+        let pixel_stride = byte_stride / byte_stride;
+        assert!(pixel_stride > width);
+        // Assumes that the user properly allocated the size for dst_buffer.
+        assert!(calculate_length(pixel_stride, height, format, pixel_type) == dst_buffer.len());
+
+        unsafe {
+            // We don't want any alignment padding on pixel rows.
+            self.ffi_gl_.PixelStorei(ffi::PACK_ALIGNMENT, 1);
+            if pixel_stride != width {
```

Comment:
should we set this unconditionally, since we don't know the current state?

Reply:
Yeah, and I guess we probably need to restore it because the other paths don't set it